src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,83 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULTL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULTL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        Node<K, V> currentNode = table[getIndex(key)];
+        if (currentNode == null) {
+            table[getIndex(key)] = new Node<>(key, value);
+            size++;
+            return;
+        }
+        while (currentNode.next != null || Objects.equals(currentNode.key, key)) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
```

Comment:
here currentNode can become null and in while you will get null pointer exception. Check currentNode in while loop